code-challenges.js

```diff
@@ -3,12 +3,29 @@
 
 // --------------------1) Create a function that takes in an array of objects and returns a sentence about each person with their names capitalized.
 
-var people = [{name: "ford prefect", occupation: "hitchhiker"}, {name: "zaphod beeblebrox", occupation: "president of the galaxy,"}, {name: "arthur dent", occupation: "radio employee"}]
+var people = [{name: "ford prefect", occupation: "hitchhiker"}, {name: "zaphod beeblebrox", occupation: "president of the galaxy"}, {name: "arthur dent", occupation: "radio employee"}]
 // Expected output example: "Ford Prefect is a hitchhiker." "Zaphod Beeblebrox is the president of the galaxy." "Arthus Dent is a radio employee."
-
-
-
-
+const capitalizeName = (arr) =>{
+    //iterates through the array to reach each object
+    arr.forEach(el => {
+        //split first and last name into an array
+        let fullNameArr = el.name.split(" ")
+        //capitalize the first letter of names
+        fullNameArr = fullNameArr.map(name => {
+            return name.charAt(0).toUpperCase() + name.substring(1)
+        })
+        //join the names to get a full name
+        let fullName = fullNameArr.join(" ")
+        //output the results
+        if (el.occupation.includes("president")){
+            console.log(`${fullName} is the ${el.occupation}.`)
+        }else {
+            console.log(`${fullName} is a ${el.occupation}.`)
+        }
+    })
+}
+
+capitalizeName(people)
```

Comment:
great job! how can we refactor this to include a return though instead of a console.log?